### PR TITLE
Wip/9551 force read only math plugin

### PR DESCRIPTION
**Pull request by d-rothe**
please consider  wip/9439-lmapParseHexInt first since I based this branch on that

### CODE REVIEW

**Comment by mhier** [comment]
This PR seems to be identical with  #288 - did something went wrong with the branches?

**Comment by d-rothe** [comment]
> This PR seems to be identical with #288 - did something went wrong with the branches?
it's because I based this branch on wip/9439-lmapParseHexInt, please look at that pull request first. But work is actually unrelated.

**Comment by mhier** [comment]
please rebase it on the master after #288 has been merged. I cannot really see what should be part of this PR otherwise... besides, the commit messages also don't really say much, please improve!

**Comment by d-rothe** [comment]
> please rebase it on the master after #288 has been merged. I cannot really see what should be part of this PR otherwise... besides, the commit messages also don't really say much, please improve!

rebase was done

**Review by mhier — CHANGES_REQUESTED** [review]
Two minor formal changes only... Please keep in mind that the commit message (with the explanations from the comment) needs to be entered when squash merging (or alternatively squash the two commits first, edit the commit message and then merge without squashing into the master)

File: tests/executables_src/testLMapForceReadOnlyPlugin.cc [hunk 1]
```diff
@@ -32,4 +32,36 @@ BOOST_AUTO_TEST_CASE(test) {
...
 
+BOOST_AUTO_TEST_CASE(test2) {
+  // this xlmap was causing a logic_error although it should not.
+  // See ticket https://redmine.msktools.desy.de/issues/9551
+  /* Reason for failure was:
+   * - proper register catalogue is only constructed in getRegisterCatalogue().
+   * - There plugins are processed in the end
+   * - but already before, register info from target register is copied as a basis for the new lmap regInfo
+   * - the wrong copy (not having included effects of target Register plugins) is never fixed
+   */
```
[the remark this answers is not in the document]
Comment (d-rothe) [reply tests/executables_src/testLMapForceReadOnlyPlugin.cc]: ok done

**Review by mhier — APPROVED** [review]

**Pull request merged**